src/gui/src/SoundLibrary/SoundLibraryPropertiesDialog.cpp

```diff
@@ -56,20 +58,104 @@ SoundLibraryPropertiesDialog::SoundLibraryPropertiesDialog( QWidget* pParent, Dr
 		authorTxt->setText( QString( drumkitInfo->get_author() ) );
 		infoTxt->append( QString( drumkitInfo->get_info() ) );
 		licenseTxt->setText( QString( drumkitInfo->get_license() ) );
+		imageText->setText( QString ( drumkitInfo->get_image() ) );
+		imageLicenseText->setText( QString ( drumkitInfo->get_image_license() ) );
+		// Licence with attribution is often too long...
+		imageLicenseText->setToolTip( QString( drumkitInfo->get_image_license() ) );
+
+		QPixmap *pixmap = new QPixmap (drumkitInfo->get_path() + "/" + drumkitInfo->get_image());
+		// scale the image down to fit if required
+		int x = (int) drumkitImageLabel->size().width();
+		int y = drumkitImageLabel->size().height();
+		float labelAspect = (float) x / y;
+		float imageAspect = (float) pixmap->width() / pixmap->height();
+
+		if ( ( x < pixmap->width() ) || ( y < pixmap->height() ) )
+		{
+			if ( labelAspect >= imageAspect )
+			{
+				// image is taller or the same as label frame
+				*pixmap = pixmap->scaledToHeight( y );
+			}
+			else
+			{
+				// image is wider than label frame
+				*pixmap = pixmap->scaledToWidth( x );
+			}
+		}
+		drumkitImageLabel->setPixmap(*pixmap);
+		drumkitImageLabel->show();
+
 	}
 
 }
 
 
-
-
 SoundLibraryPropertiesDialog::~SoundLibraryPropertiesDialog()
 {
 	INFOLOG( "DESTROY" );
 
 }
 
+void SoundLibraryPropertiesDialog::updateImage( QString& filename )
+{
+	QPixmap *pixmap = new QPixmap ( filename );
+	// scale the image down to fit if required
+	int x = (int) drumkitImageLabel->size().width();
+	int y = drumkitImageLabel->size().height();
+	float labelAspect = (float) x / y;
+	float imageAspect = (float) pixmap->width() / pixmap->height();
+
+	if ( ( x < pixmap->width() ) || ( y < pixmap->height() ) )
+	{
+		if ( labelAspect >= imageAspect )
+		{
+			// image is taller or the same as label frame
+			*pixmap = pixmap->scaledToHeight( y );
+		}
+		else
+		{
+			// image is wider than label frame
+			*pixmap = pixmap->scaledToWidth( x );
+		}
+	}
+	drumkitImageLabel->setPixmap(*pixmap);
+	drumkitImageLabel->show();
+
+}
+
+void SoundLibraryPropertiesDialog::on_imageBrowsePushButton_clicked()
+{
+	// Try to get the drumkit directory and open file browser
+	QString drumkitDir = Filesystem::drumkit_dir_search( nameTxt->text() ) + "/" + nameTxt->text();
+
+	QString fileName = QFileDialog::getOpenFileName(this, trUtf8("Open Image"), drumkitDir, trUtf8("Image Files (*.png *.jpg *.jpeg)"));
+
+	// If cancel was clicked just abort
+	if ( fileName == NULL )
+	{
+		return;
+	}
 
+	// If this file is in different directory copy it here
+	
+	QFile file( fileName );
+	QFileInfo fileInfo(file.fileName());
+	//ERRORLOG(fileInfo.dir().path().toLocal8Bit() + drumkitDir);
```

Comment:
Took a look, and maybe I'm missing what you thought may not have been intentional... or maybe not....

If it's the `// If this file is in different directory copy it here` bit - the idea is that if a user loads an image from ~/Downloads/ or something, it will copy it into the drumkit's directory.

I just had a quick look, and I'm having that "I'm missing something obvious" feeling.... let me know. ;)